src/Main.py

```diff
@@ -0,0 +1,171 @@
+        question = quiz.new_question(Screenshot.extract_question(screenshot))
+
+        # Extract all three answers from the screenshot, in three different threads
+        future_answers = {
+            pool.submit(
+                Screenshot.extract_answer,
+                screenshot,
+                question,
+                position
+            ): position for position in range(3)}
+        future_question = pool.submit(Sanitize.clean_question(question))
+        # Wait until all threads are done
+        for future in as_completed(future_answers): pass
+
+        print(f"\nQuestion n.{i}: {question.get_text()}")
+        print(f"Answers: [{question.get_answer(0).get_text()}, {question.get_answer(1).get_text()}, {question.get_answer(2).get_text()}]")
+
+        # Briefly ... later
+
+        # Define the query URL
+        query_url = Scraping.define_url(question.get_text())
+        # Get search results from Google 
+        google_results = Scraping.search(query_url)
+
+        # Parallelize the pattern matching process with all the results
+        future_guess = {
+            pool.submit(
+                Scraping.guess_answer,
+                google_results,
+                question.get_answer(position)
+            ): position for position in range(3)
+        }
+        
+        # If at least one result was found
+        for future in as_completed(future_guess):
+            if future.result():
+                question.one_match = True
+
+        # If at least one answer has a match
+        if question.one_match:
+            # Get and print the answer with the highest matches number
+            guessed = question.get_answer_max_matches()
+            print(f"\n{Style.BRIGHT}{Fore.GREEN}{guessed.get_text():>40} {Fore.CYAN}{guessed.get_matches():<40}{Fore.RESET}{Style.RESET_ALL}")
+            # Save a reference to the guessed answer (non-zero indexed)
+            question.set_guessed_answer(question.answers.index(guessed)+1)
+        else:
+            print(f"{Style.DIM}{Fore.YELLOW}No match found, trying a more in depth analysis...{Fore.RESET}{Style.RESET_ALL}\n")
+            # Perform a query built concatenating the question and each answer
+            
+            # Print sort-of table header (17+Answer+17, Score+5, 1+Results+2, Total)
+            # I know it's ugly, maybe I'll use Rich lib 
+            print(f"{Style.BRIGHT}                 Answer                 Score      Results  Total{Style.RESET_ALL}")
+
+            # Check if this is an usual question or a "negated" question (explaination below)
+            question.usual_question = "NON" not in question.get_text()
+
+            # Parallelize the pattern matching process with all the results
+            future_calculation = {
+                pool.submit(
+                    Scraping.calculate_concat,
+                    question.get_text() if question.usual_question else question.get_text().replace("NON",''),
```

Comment:
NON? considera di spostarlo per un eventuale nazionalizzazione. In generale non usare stringhe hardcoded cosi' nel codice